go-staticcheck.toml

```diff
@@ -81,6 +81,16 @@ rewrite='!strings.Contains(:[1], :[2])'
 match='strings.Index(:[1], :[2]) < 0'
 rewrite='!strings.Contains(:[1], :[2])'
 
+############################################
+# https://staticcheck.io/docs/checks#S1002 #
+############################################
+[S1002_01]
+match='if :[1] == true'
+rewrite='if :[1]'
+
+[S1002_02]
+match='if :[1] == false'
+rewrite='if !:[1]'
```

Comment:
There could be false positives with `if :[1] == true` and `if :[1] == false`, like in this code:

```go
if derp == trueVar
```

There is also trouble with these sorts of patterns if it matches previous expressions, and tries to negate them:

```go
if a == b && t == false
```

becomes

```go
if !a == b && t
```

which is not correct.

---

I suggest making the patterns more precise, which may miss some cases, but at least it will not generate false positives or incorrect code. For example:

```
match='if :[1:e] == true '
```

(with space after `true`, you could also try a regex to not match a word character, etc.). Just an example where this pattern is is guaranteed to be correct (if it matches) because `:[1:e]` can only match an expression-like syntax like `a`, `b.c`, `q(a, b)`, but not `a && b`. But, this pattern is not good enough to simplify a case like `if foo && x == true`.

Reply:
updated to basics. please confirm that rewrite syntax stays `:[1]`